lib/bin/lwm2m_carrier/os/lwm2m_os.c

```diff
@@ -215,7 +214,7 @@ int lwm2m_os_timer_start(void *timer, int32_t timeout)
 
 	timeout = get_timeout_value(timeout, work);
 
-	return k_delayed_work_submit(&work->work_item, K_MSEC(timeout));
+	return k_work_reschedule(&work->work_item, K_MSEC(timeout));
```

Comment:
FYI: This can now return 0,1 and 2. Can cause issues if user is checking "if (err)"

Reply:
aha! I remember that now, thanks

Reply:
Fixed